### PR TITLE
feat: web api to manage profile docker images

### PR DESCRIPTION
**Pull request by mswertz**
Use swagger UI to try.

Todo:
- [x] ability to white list operations (store as label on docker image deploy)
- [x] refactor test to be integration test
- [x] setting to disable docker management; then users must manage via docker-compose and config file (old way)
- [x] ability to start stopped images => currently requires update
- [x]  support images on external machines, not localhost (then also not using local docker mgt)
- [x] seperate ProfileConfig for input and output (last incl status)

BREAKING CHANGE

the Rserve configs have been merged with DataShield configs.
(with some hard work we might be able to create magic to make it non-breaking but I think this is an improvement, please feedback)

### CODE REVIEW

**Comment by mswertz** [comment]
[Implement API to manage profiles](https://trello.com/c/hjiXHQcg/918-implement-api-to-manage-profiles)

**Comment by mswertz** [comment]
[Implement API to manage profiles](https://trello.com/c/hjiXHQcg/918-implement-api-to-manage-profiles)

**Comment by mswertz** [comment]
[Pilot: Trying to use java api to start/stop profiles in Armadillo](https://trello.com/c/AxgiNAgo/917-pilot-trying-to-use-java-api-to-start-stop-profiles-in-armadillo)

**Review by tommydeboer — CHANGES_REQUESTED** [review]
As we discussed on slack, here are some other points:

- Use system persistence instead of docker labels
- Use `application.yml` to configure initial containers:
  - Don't use docker-compose because then we need labels
  - Ops can specify default setup if they like
  - Persistence is populated with initial values
  - Devs don't need to do anything extra

**Comment by tommydeboer** [comment]
**Issue:**
1. Create a `test` profile (following the test script)
2. `GET /profiles` gives:
```json
{
  "current": "default",
  "available": [
    "default",
    "test"
  ]
}
```
3. `POST /select-profile` with "test" gives:
```json
{
  "timestamp": "2022-08-18T12:06:20.508+00:00",
  "status": 404,
  "error": "Not Found",
  "message": "Profile: \"test\" not found",
  "path": "/select-profile"
}
```

File: armadillo/src/main/java/org/molgenis/armadillo/controller/DatashieldProfileController.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    this.datashieldProfileManager = requireNonNull(datashieldProfileManager);
+  }
+
+  @Operation(
+      summary = "list profiles",
+      description = "List currently installed profiles.",
+      security = {@SecurityRequirement(name = "jwt")})
+  @GetMapping(produces = APPLICATION_JSON_VALUE)
+  public List<ProfileConfigProps> listProfiles(Principal principal) {
+    this.auditEventPublisher.audit(principal, "List armadillo profiles", Map.of());
```
[the remark this answers is not in the document]
Comment (mswertz) [reply armadillo/src/main/java/org/molgenis/armadillo/controller/DatashieldProfileController.java:57]: ah, now I understand the weird pattern of audit(function). Thanks makes sense that you can then also audit exceptions. I thought these would be captured centrally but then you don't know what the context was of the exception. Nice.

File: armadillo/src/main/java/org/molgenis/armadillo/controller/DatashieldProfileController.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    return datashieldProfileManager.listDatashieldProfiles();
+  }
+
+  @Operation(
+      summary = "add profile",
+      description = "Add a new profile",
+      security = {@SecurityRequirement(name = "jwt")})
+  @PutMapping(produces = TEXT_PLAIN_VALUE)
+  @ResponseStatus(HttpStatus.OK)
+  public void addProfile(Principal principal, @RequestBody ProfileConfigProps profileConfigProps)
```
[the remark this answers is not in the document]
Comment (mswertz) [reply armadillo/src/main/java/org/molgenis/armadillo/controller/DatashieldProfileController.java:67]: I believe these options are also necessary? They were used for the 'seed' value. So I conclude we hardcode this seed value then in new value object so we can simplify this payload, agreed?

**Review by mswertz — COMMENTED** [review]
I am happy for this code to be merged.

What remaining points do you want to change @tommydeboer ?

(I can't approve because I created the PR)

**Review by tommydeboer — APPROVED** [review]

**Pull request merged**